internal/inventory/app/routes.go

```diff
@@ -59,90 +60,95 @@ func (app *InventoryApp) addRoutes() error {
 	return nil
 }
 
-func (app *InventoryApp) addRoute(path, method string, f http.HandlerFunc) error {
-	if err := app.service.AddRoute(path, f, method); err != nil {
+func (app *InventoryApp) addRoute(path, method string, f echo.HandlerFunc) error {
+	if err := app.service.AddCustomRoute(path, false, f, method); err != nil {
 		return fmt.Errorf("failed to add route, path=%s, method=%s: %w", path, method, err)
 	}
 	return nil
 }
 
 // Routes
-func (app *InventoryApp) index(w http.ResponseWriter, req *http.Request) {
-	http.ServeFile(w, req, "static/html/index.html")
+func (app *InventoryApp) index(ctx echo.Context) error {
+	http.ServeFile(ctx.Response().Writer, ctx.Request(), "static/html/index.html")
+	return nil
 }
 
-func (app *InventoryApp) getReaders(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) getReaders(ctx echo.Context) error {
+	w := ctx.Response().Writer
 	w.Header().Set("Content-Type", "application/json")
 	if err := app.defaultGrp.WriteReaders(w); err != nil {
 		msg := fmt.Sprintf("Failed to write readers list: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
```

Comment:
Make same change to all handlers below.
```suggestion
		return ctx.String(http.StatusInternalServerError, msg)
```